hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/UnderReplicatedProcessor.java

```diff
@@ -27,9 +29,9 @@
 public class UnderReplicatedProcessor extends UnhealthyReplicationProcessor
         <ContainerHealthResult.UnderReplicatedHealthResult> {
 
-  public UnderReplicatedProcessor(ReplicationManager replicationManager,
-                                  long intervalInMillis) {
-    super(replicationManager, intervalInMillis);
+  UnderReplicatedProcessor(ReplicationManager replicationManager,
+      Supplier<Duration> interval) {
```

Comment:
What's the reason for passing in a Supplier?

Reply:
If we simply pass the current value (`long` or `Duration`), an updated config value will not be reflected in the `(Under|Over)ReplicatedProcessor`.